File: app/src/main/java/com/hades/mvvmdatabindinglivedata/utils/BindingUtils.java
package com.hades.mvvmdatabindinglivedata.utils;

import android.databinding.BindingAdapter;
import android.view.View;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

/**
 * Created by Hades on 9/10/2018.
 */
public class BindingUtils {

    @BindingAdapter("imageUrl")
    public static void setImageUrl(ImageView view, int resource) {
        Glide.with(view.getContext())
                .load(resource)
                .into(view);
    }

    @BindingAdapter("visibleGone")
    public static void showHide(View view, boolean show) {
        view.setVisibility(show ? View.VISIBLE : View.INVISIBLE);
    }


}

File: app/src/main/java/com/hades/kotlintrainning/utils/BindingUtils.kt
package com.hades.kotlintrainning.utils

import android.databinding.BindingAdapter
import android.support.v7.widget.AppCompatImageView
import android.widget.ImageView
import com.bumptech.glide.Glide
import de.hdodenhof.circleimageview.CircleImageView

@BindingAdapter(value = ["app:url","app:quality"],requireAll = false)
fun setImageViewResource(imageView: ImageView, url: String?,quality : String? = "original") {
    Glide.with(imageView).load("http://image.tmdb.org/t/p/$quality/$url").into(imageView)
}

@BindingAdapter(value = ["app:url","app:quality"],requireAll = false)
fun setImageViewResource(imageView: CircleImageView, url: String?,quality : String? = "original") {
    Glide.with(imageView).load("http://image.tmdb.org/t/p/$quality/$url").into(imageView)
}

@BindingAdapter(value = ["app:thumbnailKey","app:quality"], requireAll = false)
fun setYoutubeImageResource(imageView : AppCompatImageView, key : String?, quality: String? = "original") {
    Glide.with(imageView).load("https://img.youtube.com/vi/$key/$quality.jpg").into(imageView)
}